lbrynet/blob/blob_file.py

```diff
@@ -150,12 +144,16 @@ def is_downloading(self):
         return False
 
     def read(self, write_func):
+        """
+        This function is only used in StreamBlobDecryptor
+        and should be deprecated in favor of open_for_reading()
+        """
         def close_self(*args):
             self.close_read_handle(file_handle)
             return args[0]
 
         file_sender = FileSender()
-        reader = HashBlobReader(write_func)
+        reader = HashBlobReader_v0(write_func)
```

Comment:
Since this was the only usage of `HashBlobReader`, there doesn't need to be a new `HashBlobReader_v0` class.

Reply:
It's possible, FYI the new class is HashBlobReader , and the old class was renamed from HashBlobReadr to HashBlobReader_v0 ,

HashBlobReader_v0 provides a read interface to StreamBlobDecryptor in a producer/consumer pattern.  But StreamBlobDecryptor can be reworked to use a read /write handle interface.